### PR TITLE
[Bug fix] Stop erasing schemas of frames with no data rows

### PR DESCRIPTION
**Pull request by noncomputable**
## Description
Solving issue #2528. There was code explicitly causing the issue, with a comment motivating it:

> if no rows in current frame; append is equivalent to
> directly updating columns.

But this assumption is not true to the docs and doesn't justify erasing the existing features. Any defined features, even if they don't have rows yet, should be initialized for added nodes and edges.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [x] Related issue is referred in this PR

### CODE REVIEW

**Review by BarclayII — APPROVED** [review]
Good catch!  And thanks for the unit tests as well!

**Comment by BarclayII** [comment]
I also noticed that you have another PR #2527 which has the same code.  Should I close that one?

**Comment by noncomputable** [comment]
> I also noticed that you have another PR #2527 which has the same code. Should I close that one?

No I've just made a change to that branch after a discussion with Minjie.

**Comment by noncomputable** [comment]
And no problem, happy to help!

**Comment by jermainewang** [comment]
@noncomputable Would you please add yourself to https://github.com/dmlc/dgl/blob/master/CONTRIBUTORS.md ?

**Comment by BarclayII** [comment]
> @noncomputable Would you please add yourself to https://github.com/dmlc/dgl/blob/master/CONTRIBUTORS.md ?

As a TODO item we should scan the contributor list and add them later ourselves.

**Comment by noncomputable** [comment]
> @noncomputable Would you please add yourself to https://github.com/dmlc/dgl/blob/master/CONTRIBUTORS.md ?

Yep, got it. https://github.com/dmlc/dgl/pull/2538

File: python/dgl/batch.py [hunk 1]
```diff
@@ -62,6 +62,8 @@ def batch(graphs, ndata=ALL, edata=ALL, *,
         Node features to batch.
     edata : list[str], None, optional
         Edge features to batch.
+    include_empty_data : bool, optional
+        Whether to copy a frame even if no nodes or edges exist for it.
```
Comment (jermainewang) [inline python/dgl/batch.py:66]: remove

**Pull request merged**